benchmarks/lockhammer/src/lockhammer.c

```diff
@@ -175,10 +175,11 @@ int main(int argc, char** argv)
           case 'o':
             args.pinorder = calloc(num_cores, sizeof(int));
             if (args.pinorder == NULL) {
-                fprintf(stderr, "ERROR: Cannot allocate enough memory for pinorder structure.\n");
+                fprintf(stderr, "ERROR: cannot allocate enough memory for pinorder structure.\n");
                 return 1;
             }
-            csv = strtok(optarg, ",");
+            /* colon is better than comma because lockhammer output uses csv format */
```

Comment:
I don't see the issue here.  Using a comma to delimit a list is more natural.

Reply:
The output result file is like this if we keep using comma instead of colon:

1, 0.659050, 27.533800, 27.936200, 41.778000, 0.000000, 2018-06-21 00:51:06.954413, epyc7601, ../build/lh_cas_event_mutex, -t, 1, -a, 5000, -c, 0ns, -p, 0ns, -o, 0,32,8,40,16,48,24,56,4,36,12,44,20,52,28,60,1,33,9,41,17,49,25,57,5,37,13,45,21,53,29,61,2,34,10,42,18,50,26,58,6,38,14,46,22,54,30,62,3,35,11,43,19,51,27,59,7,39,15,47,23,55,31,63

For python csv file loader, it is a little confusing.

If we change -o comma to colon, it's much easier for the python result parser to load:
1, 0.659050, 27.533800, 27.936200, 41.778000, 0.000000, 2018-06-21 00:51:06.954413, epyc7601, ../build/lh_cas_event_mutex, -t, 1, -a, 5000, -c, 0ns, -p, 0ns, -o, 0:32:8:40:16:48:24:56:4:36:12:44:20:52:28:60:1:33:9:41:17:49:25:57:5:37:13:45:21:53:29:61:2:34:10:42:18:50:26:58:6:38:14:46:22:54:30:62:3:35:11:43:19:51:27:59:7:39:15:47:23:55:31:63

The colon sign can also be perceived as POSIX path/way separator semantically.

Reply:
I agree with both arguments: commas on the command line make for more sensible list separation but colons make parsing the result easier since we don't have to distinguish between commas that separate fields and comas that separate list entries.  Is it possible to have the output always dump colons but the input accept either colons or commas?

Reply:
OK, but accepting both colon and comma will make option parsing code more complex:

1. It needs to detect if it is colon or comma separated, or perhaps both.
2. And output result may not reflect the actual input parameters.
3. Help messages will be messy as well.

I think colon only should be ok, especially at this early stage, strict syntax is better because there is no history burden.

Reply:
This can be achieved by simply:

csv = strtok(optarg, ":,");

To allow either ':' or ',' (or any combination thereof within the argument) as a tokenizing character.

In such a case output arguments should always reflect a input option which is functionally identical to whatever input was provided.  Help messages can prefer one of the options and the code can silently allow the other.

Reply:
Done, support both comma and colon and help message only shows colon as official delimiter for -o option. Fixed and verified on arm64 system in latest patch.